ddl/partition.go

```diff
@@ -603,16 +603,26 @@ func checkDropTablePartition(meta *model.TableInfo, partLowerNames []string) err
 // removePartitionInfo each ddl job deletes a partition.
 func removePartitionInfo(tblInfo *model.TableInfo, partLowerNames []string) []int64 {
 	oldDefs := tblInfo.Partition.Definitions
-	newDefs := make([]model.PartitionDefinition, 0, len(oldDefs)-1)
-	var pids []int64
-	for _, partName := range partLowerNames {
-		for i := 0; i < len(oldDefs); i++ {
-			if oldDefs[i].Name.L != partName {
-				continue
+	newDefs := make([]model.PartitionDefinition, len(oldDefs)-len(partLowerNames))
+	j := 0
+	pids := make([]int64, len(partLowerNames))
+	k := 0
+
+	// consider using a map to probe partLowerNames if too many partLowerNames
+	for i := range oldDefs {
+		found := false
+		for _, partName := range partLowerNames {
+			if oldDefs[i].Name.L == partName {
+				found = true
+				break
 			}
-			pids = append(pids, oldDefs[i].ID)
-			newDefs = append(oldDefs[:i], oldDefs[i+1:]...)
-			break
+		}
+		if found {
+			pids[k] = oldDefs[i].ID
+			k++
```

Comment:
Why change the code？I think the original code(`pids = append(pids, oldDefs[i].ID)`) is more readable.

Reply:
addressed in https://github.com/pingcap/tidb/pull/18593/commits/1f8b3589c8740d153e116e91c81590140a0314e5